### PR TITLE
[Merge-Queue] Forbid ChangeLog modification

### PR DESCRIPTION
**Pull request by JonWBedard**
#### e1d8cbe3852fc9653b7e560f9c76cf6322c3e02c
<pre>
[Merge-Queue] Forbid ChangeLog modification
<a href="https://bugs.webkit.org/show_bug.cgi?id=239413">https://bugs.webkit.org/show_bug.cgi?id=239413</a>
&lt;rdar://problem/91839553 &gt;

Reviewed by Aakash Jain.

* Tools/CISupport/ews-build/factories.py:
(MergeQueueFactoryBase.__init__): Remove AddReviewerToChangeLog and ValidateChangeLogAndReviewer.
* CISupport/ews-build/factories_unittest.py:
(TestExpectedBuildSteps):
* Tools/CISupport/ews-build/steps.py:
(PushCommitToWebKitRepo.evaluateCommand): Remove AddReviewerToChangeLog.
(ValidateCommitMessage._files): List all modified files.
(ValidateCommitMessage.evaluateCommand): Fail step if no ChangeLog modified.
(AddReviewerToChangeLog): Deleted.
* Tools/CISupport/ews-build/steps_unittest.py:

Canonical link: <a href="https://commits.webkit.org/250643@main">https://commits.webkit.org/250643@main</a>
git-svn-id: <a href="https://svn.webkit.org/repository/webkit/trunk@294323">https://svn.webkit.org/repository/webkit/trunk@294323</a> 268f45cc-cd09-0410-ab3c-d52691b4dbfc
</pre>

### CODE REVIEW

**Comment by cdumez** [comment]
Given that ChangeLog files to not get rebased properly with recent SDKs, I think this is a terrible idea.

**Comment by cdumez** [comment]
Let's first fix changelogs rebasing instead of forcing people into a broken workflow.

**Comment by cdumez** [comment]
See rdar://92962066

File: Tools/CISupport/ews-build/steps.py [hunk 1]
```diff
@@ -5015,9 +5021,11 @@ def run(self, BufferLogObserverClass=logobserver.BufferLogObserver):
             return rc
 
         log_text = self.log_observer.getStdout().rstrip()
-        if log_text:
-            self.summary = log_text
+        if any(['ChangeLog' in file for file in self._files()]):
+            self.summary = 'ChangeLog modified, WebKit only allows commit messages'
```
Comment (aj062) [inline Tools/CISupport/ews-build/steps.py:4976]: Might reword it slightly. Maybe: 'ChangeLog modification found, this is not allowed anymore, please use commit message'

**Review by aj062 — APPROVED** [review]

**Comment by aj062** [comment]
Love the title change:
`JonWBedard changed the title "[Merge-Queue] Require ChangeLog modification" to "[Merge-Queue] Forbid ChangeLog modification"`

(Too bad that GitHub doesn't allow commenting on these events)

**Comment by aj062** [comment]
Please look at unit tests failures.

**Pull request merged**